src/main/java/wolox/training/models/User.java

```diff
@@ -0,0 +1,93 @@
+package wolox.training.models;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+@Entity
+@Table(name = "users")
+public class User {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private long id;
+
+  @Column(nullable = false)
+  private String username;
+  @Column(nullable = false)
+  private String name;
+  @Column(nullable = false)
+  private LocalDate birthDate;
+
+  @Column(nullable = false)
+
+  @ManyToMany(cascade = CascadeType.ALL)
```

Comment:
This cascade type is wrong.
By using ALL, you mean: Persist, Remove, Detach, Merge, Refresh
Detach is never used mecause it means to cascade a manual detach of a model from it's persistence context, which I don't recall having used it EVER.
Persist means, anything I add to this collection, should be persisted as an entity itself. Books live independantly of their association to one or more users, so we don't need this.
Remove, analogous to Persist.
Merge: Changes to this collection should be "flushed" to the DB
Refresh: Changes in the DB should be "brought" to memory.

Therefore, you want only Merge and Refresh, given the only thing you want is to keep the Join Table updated :)

Reply:
That makes sense!